chainntnfs/btcdnotify/btcd_test.go

```diff
@@ -0,0 +1,824 @@
+// +build !race
+
+// See NOTE below as to why we disable the race detector for these tests.
+
+package btcdnotify
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/roasbeef/btcd/btcjson"
+	"github.com/roasbeef/btcd/chaincfg"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/integration/rpctest"
+	"github.com/roasbeef/btcd/txscript"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+var (
+	testPrivKey = []byte{
+		0x81, 0xb6, 0x37, 0xd8, 0xfc, 0xd2, 0xc6, 0xda,
+		0x63, 0x59, 0xe6, 0x96, 0x31, 0x13, 0xa1, 0x17,
+		0xd, 0xe7, 0x95, 0xe4, 0xb7, 0x25, 0xb8, 0x4d,
+		0x1e, 0xb, 0x4c, 0xfd, 0x9e, 0xc5, 0x8c, 0xe9,
+	}
+
+	privKey, pubKey = btcec.PrivKeyFromBytes(btcec.S256(), testPrivKey)
+	pubKeyAddr, _   = btcutil.NewAddressPubKey(pubKey.SerializeCompressed(),
+		&chaincfg.RegressionNetParams)
+	testAddr = pubKeyAddr.AddressPubKeyHash()
+)
+
+// testConfDetailsFromTxIndex tests that we can find the transaction from the
+// node's transaction index and that the details retrieved are correct.
+func testConfDetailsFromTxIndex(t *testing.T, miner *rpctest.Harness,
```

Comment:
Are these similar to those for `bitcoind`? Possible to put the two notifiers behind a common interface, and reuse the tests?